enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,428 @@
+
+### Goals
+
+- All nodes created during the installation are configured with static IPs.  
+  Rationale: Many environments, due to security policies, do not allow DHCP.
+
+- The IPI installation method is able to provide static IPs to the nodes
+  Rationale: Some users must qualify each tool used in their environment. 
+  Leveraging IPI greatly reduces the number of tools required to provision
+  a cluster.
+
+### Non-Goals
+
+- OpenShift will not be responsible for managing IP addresses.
+
+## Proposal
+
+### Static IPs Configured at Installation
+
+To faciliate the configuration of static IP address, network device configuration definitions are created for each node in the install-config.yaml. A `hosts` slice will be introduced to the installer platform specification to allow network device configurations to be specified for a nodes. 
+If a `hosts` slice is defined, enough `hosts` must be defined to cover the bootstrap, control plane, and compute nodes to be provisioned during the installation process.
+
+The bootstrap and control plane nodes will have their static IP addresses provisioned by terraform.  Compute nodes will rely on the machine API to provision their IP addresses.
+
+Example of platform specification with IP addresses defined for bootstrap, control plane, and compute nodes:
+~~~yaml
+platform:
+  vsphere:
+    hosts:
+    - role: bootstrap
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.240/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1a
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.241/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1b
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.242/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: control-plane
+      failureDomain: us-east-1c
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.243/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.244/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.245/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+    - role: compute
+      networkDevice:
+        ipAddrs:
+        - 192.168.101.246/24
+        gateway4: 192.168.101.1
+        nameservers:
+        - 192.168.101.2
+~~~
+
+### Scaling in Additional Machines
+
+Additional machines requiring static IP addresses may be scaled in by:
+- An OpenShift administrator prepares a machine yaml with the static IP address included in the provider specification
```

Comment:
So we are still going with the ability to directly provision machines with static IPs? Is this what is done upstream? I had assumed every static IP would come through an `IPAddressClaim`

Reply:
Yes, static IPs may be directly provided in the machine spec upstream as well.